.github/actions/setup-job/action.yml

```diff
@@ -19,14 +22,43 @@ runs:
     - name: Install just
       uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb #v1.4.0
       with:
-        just-version: '0.10.5'
+        just-version: ${{ env.JUST_VERSION }}
+
+    - name: Restore Rust toolchain cache
+      uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
+      if: inputs.rust != 'false'
+      with:
+        working-directory: ${{ env.RUST_WORKSPACE }}
+
+    - name: Restore Rust build cache
+      uses: actions/cache@v2
+      if: inputs.rust != 'false'
+      with:
+        path: |
+          target/
```

Comment:
This is already cached by `rust-cache`.
We can move this call at the end together with the other rust parts.

Reply:
Mmh, I'm going to double-check then. I saw some build times cut in half with this caching added, but maybe that was just a fluke.

Reply:
`rust-cache` does not keep our own build code: https://github.com/Swatinem/rust-cache/issues/37 
The code is not versioned for the cache so this can create problem on the long run.

Reply:
Sure, but the cache here is bound to the Cargo.lock, which changes quite frequently for us. So for branches I definitely caching `target/` is desirable, though one could think about disabling it in bors / main.

Reply:
In any case, I'm going to revert that part to the state of before for now. No need to look into this in detail at the moment.